instrumentation-tests/src/androidTest/java/com/mapbox/navigation/instrumentation_tests/core/ClosuresTest.kt

```diff
@@ -0,0 +1,238 @@
+        latitude = origin.latitude()
+        longitude = origin.longitude()
+    }
+
+    @Before
+    fun setUp() {
+        runOnMainSync {
+            mapboxNavigation = MapboxNavigationProvider.create(
+                NavigationOptions.Builder(context)
+                    .accessToken(getMapboxAccessTokenFromResources(context))
+                    .routingTilesOptions(
+                        RoutingTilesOptions.Builder()
+                            .tilesBaseUri(URI(mockWebServerRule.baseUrl))
+                            .build()
+                    )
+                    .build()
+            )
+        }
+    }
+
+    @Test
+    fun closuresAppearedAfterRefresh() = sdkTest {
+        // no closures in the primary route
+        setUpRequestHandler(R.raw.route_response_route_refresh)
+        mockWebServerRule.requestHandlers.add(
+            FailByRequestMockRequestHandler(
+                MockDirectionsRefreshHandler(
+                    "route_response_route_refresh",
+                    // [1, 3] closure on leg#0
+                    readRawFileText(context, R.raw.route_response_route_refresh_annotations),
+                    acceptedGeometryIndex = 0
+                )
+            )
+        )
+
+        mapboxNavigation.startTripSession()
+        stayOnPosition(origin)
+        mapboxNavigation.flowLocationMatcherResult().filter {
+            abs(it.enhancedLocation.latitude - origin.latitude()) < 0.0005 &&
+                abs(it.enhancedLocation.longitude - origin.longitude()) < 0.0005
+        }
+        val routes = requestRoutes(listOf(origin, destination)).take(1)
+        assertTrue(routes.first().directionsRoute.legs()!!.first().closures().isNullOrEmpty())
+        mapboxNavigation.setNavigationRoutesAndWaitForUpdate(routes)
+
+        val firstRouteProgress = mapboxNavigation.routeProgressUpdates()
+            .filter { it.currentState == RouteProgressState.TRACKING }
+            .first()
+
+        assertFalse(firstRouteProgress.hasUnexpectedUpcomingClosures())
+
+        mapboxNavigation.routeRefreshController.requestImmediateRouteRefresh()
+
+        mapboxNavigation.routesUpdates()
+            .filter { it.reason == RoutesExtra.ROUTES_UPDATE_REASON_REFRESH }
+            .first()
+
+        val routeProgressAfterRefresh = mapboxNavigation.routeProgressUpdates()
+            .filter { it.currentState == RouteProgressState.TRACKING }
+            .take(3).toList().last()
+
+        assertTrue(routeProgressAfterRefresh.hasUnexpectedUpcomingClosures())
+
+        // after closure [1, 3] start
+        stayOnPosition(routes.first().directionsRoute.completeGeometryToPoints()[2])
+        val routeProgressInsideClosure = mapboxNavigation.routeProgressUpdates()
+            .filter { it.currentLegProgress?.geometryIndex in 1..3 }
+            .first()
+
+        assertFalse(routeProgressInsideClosure.hasUnexpectedUpcomingClosures())
+    }
+
+    @Test
+    fun hasExpectedClosures() = sdkTest {
```

Comment:
It seems like this test is the same as the first, exept for the part where closure moves. Maybe it makes sense to move the unique check to the first test?

Reply:
No, it's not. In the test above there were no closures in the original route, while here there is one.